tmt/base.py

```diff
@@ -3978,11 +3978,13 @@ def load(self) -> None:
         plan_options = ['names', 'filters', 'conditions', 'links', 'default']
         if not any(Plan._opt(option) for option in plan_options):
             assert self.tree is not None  # narrow type
-            self._plans = [
-                plan
-                for plan in self.tree.plans(run=self)
-                if data.plans and plan.name in data.plans
-            ]
+
+            if data.plans is None:
+                plan_names = []
+            else:
+                plan_names = [f"^{plan_name}$" for plan_name in data.plans]
```

Comment:
If `plan_name` is indeed plan *name*, we need `re.escape(plan_name)`, to correctly match names like `/my.*awesome.plan`, not selecting plans like `/my.notsoawesome.plan`. `plan_name` needs to become `/my\.\*awesome\.plan`.

Reply:
Good point, fixed in bdd056bb.